modules/library/main/src/main/java/org/geotools/data/sort/SimpleFeatureIO.java

```diff
@@ -268,4 +316,17 @@ public void close(boolean deleteFile) throws IOException {
     public String toString() {
         return "SimpleFeatureIO [schema=" + schema + ", file=" + file + "]";
     }
+
+    private Collection<String> split(String value, int charSize) {
+        final AtomicInteger atomicInteger = new AtomicInteger(0);
+        Collection<String> result =
```

Comment:
I'm sure it works in today's JVMs, but it might break later, see:

https://stackoverflow.com/questions/37277967/in-java-8-does-a-sequential-and-ordered-stream-guarantee-performing-operations

I suggest to use a common iteration loop, which guarantees predictable execution order.

Reply:
Fixed, thanks.